### PR TITLE
Change the template type from arma::mat to MatType in NS

### PR DESCRIPTION
**Pull request by shrit**

### CODE REVIEW

**Review by rcurtin — COMMENTED** [review]
Thanks, want to add a super basic test using `KNN` with a `float` element type?  The changes themselves look great to me, it would just be nice to test them also. :+1:

File: src/mlpack/tests/knn_test.cpp [hunk 1]
```diff
@@ -16,6 +16,18 @@
...
+ * A couple of handful declarations for float32 testing.
+ * These will be removed when we refactor the Bounds to accept MatType.
+ * For now, we will keep the following declarations.
+ */
+template<typename MetricType>
+using FloatHRectBound = HRectBound<MetricType, float>;
+
+template<typename MetricType, typename StatisticType, typename MatType>
+using FloatKDTree = BinarySpaceTree<MetricType, StatisticType, MatType,
+                                    FloatHRectBound, MidpointSplit>;
```
Comment (rcurtin) [inline src/mlpack/tests/knn_test.cpp:29]: I am trusting that you will come back later and refactor the bounds :smile:

**Review by rcurtin — APPROVED** [review]
Looks good to me, thanks for adding the test! :+1:

File: src/mlpack/tests/knn_test.cpp (hunk 1 above)
Comment (shrit) [reply src/mlpack/tests/knn_test.cpp:29]: I have to do so, but I will need some time :+1:

**Pull request merged**